src/components/forms/NetworkDevicesForm/InheritedNetworkRow.tsx

```diff
@@ -0,0 +1,137 @@
+import NetworkDeviceActionButtons from "./NetworkDeviceActionButtons";
+import { addNicDevice, addNoneDevice } from "util/formDevices";
+
+interface Props {
+  device: InheritedNetwork;
+  project: string;
+  managedNetworks: LxdNetwork[];
+  formik: InstanceAndProfileFormikProps;
+  focusNetwork: (id: number) => void;
+  removeNetwork: (deviceName: string) => void;
+  isOverridden: boolean;
+  index: number;
+}
+
+export const getInheritedNetworkRow = ({
+  device,
+  project,
+  managedNetworks,
+  formik,
+  focusNetwork,
+  removeNetwork,
+  isOverridden,
+  index,
+}: Props): MainTableRow => {
+  const readOnly = (formik.values as EditInstanceFormValues).readOnly;
+  const overrideDevice = formik.values.devices.find(
+    (t) => t.name === device.key,
+  ) as LxdNicDevice | LxdNoneDevice;
+
+  const overrideNetwork = managedNetworks.find(
+    (t) => t.name === (overrideDevice as LxdNicDevice)?.network,
+  );
+
+  return getConfigurationRowBase({
+    configuration: (
+      <b
+        className={classNames({
+          "u-text--muted": isOverridden,
+          "u-text--line-through": isOverridden,
+        })}
+      >
+        {device.key}
+      </b>
+    ),
+    inherited: (
+      <div>
+        <div
+          className={classNames("p-text--small", "u-text--muted", {
+            "u-text--line-through": isOverridden,
+          })}
+        >
+          From profile{" "}
+          <ResourceLink
+            type="profile"
+            value={device.sourceProfile}
+            to={`/ui/project/${encodeURIComponent(project)}/profile/${encodeURIComponent(device.sourceProfile)}`}
+            className={classNames({
+              "u-text--line-through": isOverridden,
+            })}
+          />
+        </div>
+        <div
+          className={classNames({
+            "u-text--muted": isOverridden,
+            "u-text--line-through": isOverridden,
+          })}
+        >
+          Network
+        </div>
+        <ResourceLink
+          type="network"
+          value={device.network?.network || ""}
+          to={`/ui/project/${encodeURIComponent(project ?? "")}/network/${encodeURIComponent(device.network?.network || "")}`}
+          className={classNames({
+            "u-text--line-through": isOverridden,
+          })}
+        />
+        <ReadOnlyAclsList
+          project={project}
+          network={managedNetworks.find(
+            (t) => t.name === device.network?.network,
+          )}
+          device={device.network}
+        />
+      </div>
+    ),
+    override: isOverridden ? (
+      <NetworkDevice
+        readOnly={readOnly}
+        formik={formik}
+        project={project}
+        focusNetwork={focusNetwork}
+        removeNetwork={removeNetwork}
+        device={overrideDevice}
+        index={index}
+        network={overrideNetwork}
+        isOverridden
+        isInheritedRow
+      />
+    ) : (
+      <div className="network-device">
+        <div />
+        <NetworkDeviceActionButtons
```

Comment:
Shall we add the detach button when an override exists? Currently, when an override exists, I have to first clear, and then I can detach the device. I think it would be nice to do that in one click.


https://github.com/user-attachments/assets/962228a1-3637-4d5e-a349-c427326bd163